sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -494,6 +494,78 @@ export class StudentService extends RecordDataModelService<Student> {
     return searchQuery.getMany();
   }
 
+  /**
+   * Search students based on the search criteria.
+   * @param searchCriteria options to search by firstName,
+   * lastName, appNumber or sin.
+   * @returns list of students.
+   */
+  async searchStudentApplicationForInstitution(
+    institutionId: number,
+    searchCriteria: {
+      firstName?: string;
+      lastName?: string;
+      appNumber?: string;
+      sin?: string;
+    },
+  ): Promise<Student[]> {
+    const searchQuery = this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.birthDate",
+        "user.firstName",
+        "user.lastName",
+        "sinValidation.sin",
+      ])
+      .innerJoin(
+        Application,
+        "application",
+        "application.student.id = student.id",
+      )
+
+      .innerJoin("application.location", "pirLocation")
+      .innerJoin("pirLocation.institution", "pirInstitution")
+
+      .leftJoin("application.currentAssessment", "studentAssessment")
```

Comment:
This one can be an inner join right as per the AC.
![image](https://user-images.githubusercontent.com/54600590/230211528-d18101b8-0815-4b17-b776-495e68fceaee.png)

Reply:
I understood the context from our conversation.